eureka2-server/src/main/java/com/netflix/eureka2/server/health/HealthStatusProviderRegistry.java

```diff
@@ -32,7 +34,8 @@ public void add(HealthStatusProvider<?> provider) {
         return providerSubject;
     }
 
-    @PostInjector
+//    @PostInjector
```

Comment:
PostInjector is a special annotation that will run annotated methods after injector is initialized. PostConstruct is non-deterministic, and the initialization may happen in any valid order. ```HealthStatusProviderRegistry``` emits events directly after the activate method is called, and if some components are not ready yet they will miss it.